Userland/Libraries/LibFileSystem/FileSystem.cpp

```diff
@@ -208,13 +208,11 @@ ErrorOr<void> copy_file(StringView destination_path, StringView source_path, str
     if (source_stat.st_size > 0)
         TRY(destination->truncate(source_stat.st_size));
 
-    while (true) {
-        auto bytes_read = TRY(source.read_until_eof());
-
-        if (bytes_read.is_empty())
-            break;
-
-        TRY(destination->write_until_depleted(bytes_read));
+    ByteBuffer buffer = TRY(ByteBuffer::create_uninitialized(1 * MiB));
+    while (!source.is_eof()) {
+        auto bytes = TRY(source.read_some(buffer));
+        TRY(destination->write_until_depleted(bytes));
+        buffer.clear();
```

Comment:
I'm pretty sure clearing the buffer is not needed. In fact, I think it would start breaking during the second loop already, because the buffer is now zero-sized.